fms/utils/generation.py

```diff
@@ -53,17 +135,35 @@ def generate(
         use_cache: requires that the model accept use_cache and
             past_key_value_states args in forward method.
     """
-    batched = False
     if num_beams != 1:
         raise NotImplementedError("generate() does yet not support beam search")
-    if type(input_ids) == torch.Tensor:
-        if input_ids.dim() != 1:
-            batched = True
-    else:
-        raise RuntimeError("generate() requires a tensor of token ids as the prefix")
 
-    if not batched:
+    if isinstance(input_ids, torch.Tensor) and len(input_ids.shape) == 1:
```

Comment:
I suspect there's a simpler way to structure these conditions. maybe as separate steps? as-is it's a bit confusing.
E.g. in the prior code `if not batched: inputs_ids = input_ids.unsqueeze(0)` was a pretty obvious two lines. Reading the same logic mixed in here is a lot less obvious and seems to want comments/explanation, but it might be possible to restructure in a way that is more obvious.

Reply:
updated in latest commit